internal/sleuth/incident_impact_source_resource.go

```diff
@@ -0,0 +1,876 @@
+package sleuth
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringdefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+
+	"github.com/sleuth-io/terraform-provider-sleuth/internal/gqlclient"
+)
+
+var (
+	_ resource.Resource                = &incidentImpactSourceResource{}
+	_ resource.ResourceWithConfigure   = &incidentImpactSourceResource{}
+	_ resource.ResourceWithImportState = &incidentImpactSourceResource{}
+)
+
+type incidentImpactResourceModel struct {
```

Comment:
So the type names start with lowercase, but attributes are uppercased?

Reply:
Type names are lowercase because they are not exported (everything lowercase in Go is only available to the package that creates it) - attributes are usually exported and are rarely created with lowercase (though it's not unseen if you would expose a cache to have private timers inside)